brave/brave-core
# Search result ad click confirmation.

Spec: https://docs.google.com/document/d/1ncuyw0Yv_qawR-FM0_e1r4pNU2pGDhS49KfwfHqqrqs/edit#heading=h.txs9x3akav5

Description is here: https://github.com/brave/security/issues/876


Resolves https://github.com/brave/brave-browser/issues/23002

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

### Test case 1
- Turn off Brave Private Ads.
- Do several search requests on search.brave.com until an ad is shown in a SERP.
- Click on the ad link.
- Check that the page navigates to https://search.anonymous.ads.brave.com/v3/click and then navigates to the ad landing page using a server redirect.

### Test case 2
- Turn on Brave Private Ads.
- Do several search requests on search.brave.com until an ad is shown in a SERP.
- Check that an ad viewed event is triggered. There should be `Viewed search result ad with placement id` message in rewards-internals log.
- Click on the ad link.
- Check that the page navigates to the ad landing page without navigation to https://search.anonymous.ads.brave.com/v3/click.
- Check that an ad clicked event is triggered. There should be `Clicked search result ad with placement id` message in rewards-internals log.
